agent/cmd/cuda-vmm-interpose/tests/test_cucheckpoint.py

```
@@ -0,0 +1,773 @@
+        raise AssertionError(
+            f"{stage}: output[{mismatch}] is {actual[mismatch].item()}, expected 3.0"
+        )
+
+
+def _visible_gpus() -> tuple[str, str]:
+    configured = os.environ.get("CUDA_VISIBLE_DEVICES")
+    if configured is None:
+        return "0", "1"
+    devices = [entry.strip() for entry in configured.split(",") if entry.strip()]
+    if len(devices) < WORLD_SIZE:
+        raise RuntimeError("CUDA_VISIBLE_DEVICES must contain two GPUs")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)